web/source/osk/visualKeyboard.ts

```diff
@@ -197,6 +211,14 @@ namespace com.keyman.osk {
         keyText=this.renameSpecialKey(tId);
       }
 
+      // Some special key codes may be used on standard keys, not just on marked 'special keys'.
+      if(spec['text'] == '*ZWNJ*') {
+        keyText = this.renameSpecialKey(spec['text']);
+        // The special OSK font is not normally used for non-special keys.
+        // We'll need a font override here.
+        spec['font'] = "SpecialOSK";
+      }
```

Comment:
Ideally, shouldn't we be doing this for *any* special text on *any* key?

Reply:
Oh.  I mean, we could... I just figured there was some legacy design decision where the answer had already been determined to be "no", since that's not what the existing code has been doing.  

The existing code has been checking for "special key" styling (`sp == '1' || sp == '2'`) as a pre-condition.

Reply:
I can't see a good case for anyone ever *wanting* to have `*foo*` on a key! So let's go ahead and do it. I know it's technically not back-compatible though -- which is something we should note in the [wiki](https://github.com/keymanapp/keyman/wiki/Keyman-13.0-to-14.0-Migration#developer-and-keyboards) -- but I don't think we need to add a developer-side check for use of these on non-special keys, do you?

Reply:
I think a "info" notification or "warning" would be appropriate, should a keyboard target versions before 14.0.  Keyboard developer shouldn't get blindsided by the keys not displaying correctly if someone loads it in a legacy version.

I don't think it's worth making an out-and-out error, as it doesn't actually _break_ a keyboard.  Just its appearance/styling.